### PR TITLE
Add grid lesson

### LINKED ISSUE #4785: Grid [Learn CSS]
[Draft](https://docs.google.com/document/d/1xwL270qcao4XOA9GPFL0G00UFSnFBwjm_W213HGkgdc/edit#)

### PR DESCRIPTION
**Pull request by rachelandrew**
Fixes #4785 

Adds the grid lesson.

### CODE REVIEW

**Review by argyleink — APPROVED** [review]
Sooooooo gooooooood!

File: src/site/content/en/learn/css/grid/index.md [hunk 1]
```diff
@@ -0,0 +1,803 @@
...
+the content starts to wrap.
+So `fit-content(10em)` will create a track that is less than 10em,
+if the `max-content` size is less than 10em,
+but never larger than 10em.
+
+In the next demo try out the different intrinsic sizing keywords by changing the sizing of the grid tracks.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'qBRqNgL'
```
Comment (una) [inline src/site/content/en/learn/css/grid/index.md:227]:
```suggestion
  id: 'qBRqNgL',
  height: 600
```

File: src/site/content/en/learn/css/grid/index.md [hunk 2]
```diff
@@ -0,0 +1,803 @@
...
+To have a component with a fixed size element and the second track taking up whatever space is left,
+you can use as a tracklisting of `grid-template-columns: 200px 1fr`.
+
+Using different values for the fr unit will share the space in proportion.
+Larger values getting more of the spare space.
+In the demo below change the value of the third track.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'vYgyXNE'
```
Comment (una) [inline src/site/content/en/learn/css/grid/index.md:264]:
```suggestion
  id: 'vYgyXNE',
  height: 600
```

File: src/site/content/en/learn/css/grid/index.md [hunk 3]
```diff
@@ -0,0 +1,803 @@
...
+You need to define row tracks otherwise the items will create intrinsic column tracks,
+and layout out all in one long row.
+
+These values relate to the writing mode of the document.
+A row always runs in the direction a sentence runs in the writing mode of the document or component.
+In the next demo you can change mode the value of `grid-auto-flow` and the `writing-mode` property.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'PoWbWbr'
```
Comment (una) [inline src/site/content/en/learn/css/grid/index.md:398]:
```suggestion
  id: 'PoWbWbr',
  height: 600
```

File: src/site/content/en/learn/css/grid/index.md [hunk 4]
```diff
@@ -0,0 +1,803 @@
...
+The next demo shows this behavior.
+The checkbox will apply the dense packing mode.
+This is enabled by giving `grid-auto-flow` a value of `dense`.
+With this value in place,
+grid will take items later in the layout and use them to fill gaps.
+This may mean that the display becomes disconnected from the logical order.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'ZELBLrJ'
```
Comment (una) [inline src/site/content/en/learn/css/grid/index.md:442]:
```suggestion
  id: 'ZELBLrJ',
  height: 600
```

File: src/site/content/en/learn/css/grid/index.md [hunk 5]
```diff
@@ -0,0 +1,803 @@
...
+
+Chrome DevTools can give you a visual guide to the lines in order to check where your item is placed.
+
+The line numbering follows the writing mode and direction of the component.
+In the next demo change the writing mode or direction
+to see how the placement of the items stays consistent to the way that text flows.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'QWdGdzd'
```
Comment (una) [inline src/site/content/en/learn/css/grid/index.md:492]:
```suggestion
  id: 'QWdGdzd',
  height: 600
```

File: src/site/content/en/learn/css/grid/index.md [hunk 6]
```diff
@@ -0,0 +1,803 @@
...
+
+Using line-based positioning you can place items into the same cell of the grid.
+This means that you can stack items,
+or cause one item to partly overlap another.
+Items which come later in the source will be displayed on top of items that come earlier.
+You can change this stacking order using `z-index` just as with positioned items.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'BapQWQW'
```
Comment (una) [inline src/site/content/en/learn/css/grid/index.md:505]:
```suggestion
  id: 'BapQWQW',
  height: 600
```

File: src/site/content/en/learn/css/grid/index.md [hunk 7]
```diff
@@ -0,0 +1,803 @@
...
+In the next example I have created a two column layout which moves to three columns
+by redefining the value of `grid-template-columns` and `grid-template-areas`.
+Open the example in a new window to play with the viewport size and see the layout change.
+
+You can also see how the `grid-template-areas` property relates to `writing-mode` and direction,
+as with other grid methods.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'oNBYepg'
```
Comment (una) [inline src/site/content/en/learn/css/grid/index.md:685]:
```suggestion
  id: 'oNBYepg',
  height: 600
```

File: src/site/content/en/learn/css/grid/index.md [hunk 8]
```diff
@@ -0,0 +1,803 @@
...
+
+### Distributing extra space
+
+In this demo the grid is larger than the space needed to lay out the fixed width tracks.
+This means we have space in both the inline and block dimensions of the grid.
+Try different values of `align-content` and `justify-content` to see how the tracks behave.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'rNjjMVd'
```
Comment (una) [inline src/site/content/en/learn/css/grid/index.md:764]:
```suggestion
  id: 'rNjjMVd',
  height: 650
```

File: src/site/content/en/learn/css/grid/index.md [hunk 9]
```diff
@@ -0,0 +1,803 @@
...
+the initial value will be `start` rather than `stretch` to avoid stretching things out of shape.
+{% endAside %}
+
+In the demo change the values of `justify-items` and `align-items` to see how this changes the layout.
+The grid area is not changing size,
+instead the items are being moved around inside the defined area.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'rNjjMVd'
```
Comment (una) [inline src/site/content/en/learn/css/grid/index.md:792]:
```suggestion
  id: 'rNjjMVd',
  height: 650
```

**Review by una — COMMENTED** [review]
This one looks great! I just added some adjustments to the height of the demos to make them a bit easier to see in this context

**Comment by argyleink** [comment]
assessments for this PRed here https://github.com/rachelandrew/web.dev/pull/1

**Review by una — APPROVED** [review]

**Pull request merged**